tiledb/sm/c_api/tiledb.h

```diff
@@ -4594,6 +4605,7 @@ TILEDB_EXPORT int32_t tiledb_array_open(
  * @note If the same array object is opened again without being closed,
  *     an error will be thrown.
  * @note This function is applicable only to read queries.
+ * @note The config should be set before opening an array.
  */
 TILEDB_EXPORT int32_t tiledb_array_open_at(
```

Comment:
Shouldn't this be deprecated?

Reply:
The problem with deprecating C APIs is that if they're used in the implementation in the C++ APIs, the compiler will complain any time that the C++ APIs are imported. We've had this problem in the past and decided to remove the macro from the C APIs, but leave it in the related C++ APIs. This is a larger design problem we can tackle later.

For example, consider that C API `tiledb_bar` and C++ API `Foo::bar` are both deprecated. If the C++ API is implemented as:
```
void Foo::bar() {
  tiledb_bar();
}
```

The will get deprecation errors just by including the C++ header for `Foo`, even if they don't use the deprecated API.